modules/kms/main.tf

```
@@ -1,15 +1,78 @@
+locals {
+  account_id        = data.aws_caller_identity.current.account_id
+  account_principal = "arn:${data.aws_partition.current.partition}:iam::${local.account_id}:root"
+
+  principals = sort(distinct(concat(
+    var.allowed_principal_arns,
+    module.allowed_role_map.principals,
+  )))
+}
+
+data "aws_caller_identity" "current" {}
+
+data "aws_partition" "current" {}
```

Related to https://github.com/cloudposse/terraform-aws-components/pull/1136/files#r1785627233

Here is an example of similar data sources turned off if the `enabled` flag is false

https://github.com/cloudposse/terraform-aws-components/blob/f51b6f31b196f1cfa3befe7f17024b1957bb8eeb/modules/aws-inspector2/main.tf#L20-L22

```suggestion
data "aws_caller_identity" "current" {
  count = module.this.enabled ? 1 : 0
}

data "aws_partition" "current" {
  count = module.this.enabled ? 1 : 0
}
```

---

Ignored